src/playground/sample_data_ingestion/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,345 @@
+        raise ValueError("You must have a template and an index name to continue")
+    if not client:
+        raise ValueError("Client needed to connect")
+    if avg_min < abs_min or avg_max > abs_max or avg_min > avg_max or abs_min > abs_max:
+        raise ValueError("Check ranges for generating trend data")
+    if anomaly_percentage < 0 or anomaly_percentage > 1:
+        raise ValueError("Anomaly percentage cannot be outside a percent")
+
+
+def build_request_body(index_name, file_provided, timestamp, minutes, chunk, chunk_index, current_index, dataset, max_bulk_size):
+    """
+    Function that constructs the request body for BULK API call
+
+    Arguments:  
+        - index_name: The name of the index to ingest data
+        - file_provided: Boolean flag as to whether or not a file was provided
+        - chunk: How many documents can be ingested per BULK call
+        - timestamp: The field name that contains timestamps
+        - minutes: The time interval for each data point (e.g. if minutes = 2, this tool will generate entries with timestamps that are 2 minutes apart from one another)
+        - current_date: The date at which entries are generated
+        - dataset: The list containing the data to ingest
+        - max_bulk_size: The max amount in bytes of a bulk call
+
+    Returns:
+        - A tuple containing the request body and the current index to look at
+    """
+
+    one_week_ago = datetime.now() - timedelta(days = 7)
+    add_to_request = []
+    request_body = []
+    current_body_size = sys.getsizeof(request_body)
+    while current_index < min(chunk_index + chunk, len(dataset)):
+        # Adds the index name
+        index_name_body = {"index": {"_index": index_name}}
+
+        # If timestamps were a field type for user provided data, modify time
+        if file_provided and timestamp:
+            one_week_ago += timedelta(minutes = minutes)
+            offset = one_week_ago.strftime("%s")
+            dataset[current_index][timestamp] = int(offset)
+        
+        # Adds the action to take
+        index_action = dataset[current_index]
+
+        # Adds both index name body and index action body to the test request body
+        add_to_request.append(index_name_body)
+        add_to_request.append(index_action)
+        current_body_size = sys.getsizeof(add_to_request)
+
+        # Adds to request only if the body size is < the max bulk size (user defined)
+        if current_body_size < max_bulk_size:
+            request_body.append(index_name_body)
+            request_body.append(index_action)
+        else:
+            break
+        current_index+= 1
+    
+    return (request_body, current_index)
+
+
+def ingest(client, 
+    template,
+    index_name,
+    file_provided = False,  
+    mapping = True,  
+    number = 6, 
+    chunk = 5, 
+    timestamp = None,
+    feature = None,  
+    avg_min = 0, 
+    avg_max = 0, 
+    abs_min = 0, 
+    abs_max = 0, 
+    anomaly_percentage = 0, 
```

Comment:
couple functions use this repeatly, is it possile to create a Custom Data Type? sample https://realpython.com/lessons/custom-data-types/

Reply:
I modified the args to include `anomaly_detection_trend`, which is a `dict` containing all the args necessary to generate a data trend that I defined in `sample_data_plugins/ad_plugin_data_config/data_trend_functions.py`